code/Helper/Entity/Producthelper.php

```diff
@@ -139,8 +136,13 @@ public function getProductCollectionQuery($storeId, $productIds = null, $only_vi
                 $catalogInventory_stock->addInStockFilterToCollection($products);
             }
 
-            $products = $products->addAttributeToSelect('special_from_date')->addAttributeToSelect('special_to_date')
-                             ->addAttributeToFilter('status', Mage_Catalog_Model_Product_Status::STATUS_ENABLED);
+            // $products = $products->addFinalPrice();
```

Comment:
@maxiloc I checked and `addFinalPrice()` method only adds some new `price` attributes to `Product` objects. But do not overrides `price` attribute. And we use only `price` attribute for inidicing. So according my research we should not need to call `addFinalPrice()` at all.
And I couldn't find a usecase where we would need it.

Did I miss something?